ui/component/or-dashboard-builder/src/or-dashboard-settingspanel.ts

```diff
@@ -275,60 +302,327 @@ export class OrDashboardSettingsPanel extends LitElement {
+                                                                            @click="${() => { this.removeThreshold(this._config, threshold); }}">
+                                                                        <or-icon icon="close-circle"></or-icon>
+                                                                    </button>
+                                                                </div>
+                                                            `
+                                                        })}
+                                                        <or-mwc-input .type="${InputType.BUTTON}" label="${i18next.t('threshold')}" icon="plus"
+                                                                      style="margin-top: 24px; margin-left: -7px;"
+                                                                      @or-mwc-input-changed="${() => this.addNewThreshold(this._config)}">
+                                                        </or-mwc-input>                                                                                                                        </div>
+                `:null}
+             </div> `
+
         }
     }
 
-    removeThreshold(config: any, threshold: [number, string]) {
-        config.thresholds = (config.thresholds as [number, string][]).filter((x) => x != threshold);
+    removeThreshold(config: any, threshold: [number, string] | [string, string]) {
+        if(typeof threshold[0] == 'number'){
+            config.thresholds = (config.thresholds as [number, string][]).filter((x) => x != threshold);
+        }
+        else {
+            config.textColors = (config.textColors as [string, string][]).filter((x) => x != threshold);
+        }
         this.updateParent(new Map<string, any>([["config", this._config]]));
     }
-    addThreshold(config: any, threshold: [number, string]) {
-        (config.thresholds as [number, string][]).push(threshold);
+    addThreshold(config: any, threshold: [number, string] | [string, string]) {
+        if(typeof threshold[0] == 'number'){
+            (config.thresholds as [number, string][]).push(threshold as [number, string]);
+        }
+        else {
+            (config.textColors as [string, string][]).push(threshold as [string, string]);
+        }
         this.updateParent(new Map<string, any>([["config", this._config]]));
     }
     addNewThreshold(config: any) {
-        const suggestedValue = (config.thresholds[config.thresholds.length - 1][0] + 10);
-        this.addThreshold(config, [(suggestedValue <= config.max ? suggestedValue : config.max), "#000000"]);
+        if(config.valueType == 'string'){
+            this.addThreshold(config, ["new", "#000000"]);
+        }
+        else if(config.deltaFormat){
+            const suggestedValue = (config.thresholds[config.thresholds.length - 1][0] + 10);
+            this.addThreshold(config, [(suggestedValue <= config.max ? suggestedValue : config.max), "#000000"]);
```

Comment:
Instead of having a different `else` statement, you can add an undefined check for `config.max` here.
```
this.addThreshold(config, [(!config.max || suggestedValue <= config.max ? suggestedValue : config.max), "#000000"]);
```